src/sage/groups/braid.py

```diff
@@ -1501,7 +1500,8 @@ def left_normal_form(self, algorithm='libbraiding'):
             return tuple([B.delta()**l[0][0]] + [B(b) for b in l[1:]] )
         elif algorithm == 'artin':
             return FiniteTypeArtinGroupElement.left_normal_form.f(self)
-
+        else:
+            raise ValueError("invalid algorithm")
```

Comment:
```suggestion
        raise ValueError("invalid algorithm")
```
This is just my personal coding style, but I think it makes it a bit more apparent (and catches more bugs) if it is just a fall through. That way the method will not secretly return `None` and each branch of the `if` statement returns something.